### PR TITLE
Add migration to `replace-border-declarations`

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Part of #7212 and #7210 

### WHAT is this pull request doing?

Adding a new `replace-border-declarations` migration script.

Running ```npx @shopify/polaris-migrator replace-border-declarations "**/*.scss"``` will target the following `border` properties and replace any `px`, `rem`, `rem()`, `border()`, `border-width()`, or `border-radius()` values that align with our with our [border tokens](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/token-groups/shape.ts#L49-L63), [border-width tokens](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/token-groups/shape.ts#L34-L48), or [border-radius tokens](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/token-groups/shape.ts#L4-L33). 

- `border` 
- `border-top`
- `border-right`
- `border-bottom`
- `border-left`
- `border-width`
- `border-top-width`
- `border-right-width`
- `border-bottom-width`
- `border-left-width`
- `border-radius`
- `border-top-left-radius`
- `border-top-right-radius`
- `border-bottom-left-radius`
- `border-bottom-right-radius`

Co-Authored-By: Chloe Rice [18447883+chloerice@users.noreply.github.com](mailto:18447883+chloerice@users.noreply.github.com)

### CODE REVIEW

File: polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+                return;
+              }
+
+              node.value = `var(${borderRadiusLengthMap[valueInPx]})`;
+            }
+
+            targets.at(-1)!.replaced = true;
+
+            return;
+          }
```
Comment (chloerice) [inline polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts:111]: This covers "the happy path" (hard coded `px`, `rem`) for both `--p-border-width-X` and `--p-border-radius-X` token mapping.

File: polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                return;
+              }
+
+              node.value = `var(${borderWidthLengthMap[valueInPx]})`;
+            }
+
+            targets.at(-1)!.replaced = true;
+
+            return;
+          }
```
Comment (chloerice) [inline polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts]: This covers "the happy path" (hard coded `px`, `rem`) for both `--p-border-width-X` and `--p-border-radius-X` token mapping.

File: polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+                node.value = 'var';
+                node.nodes = [
+                  {
+                    type: 'word',
+                    value: borderWidthLengthMap[valueInPx],
+                    sourceIndex: node.nodes[0]?.sourceIndex ?? 0,
+                    sourceEndIndex: borderWidthLengthMap[valueInPx].length,
+                  },
+                ];
+              }
```
Comment (chloerice) [inline polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts]: This covers `rem()` mapping for both `--p-border-width-X` and `--p-border-radius-X` tokens.

File: polaris-migrator/README.md [hunk 4]
```diff
@@ -121,6 +121,34 @@ Replace legacy Typography functions and hardcoded lengths with Polaris custom pr
 npx @shopify/polaris-migrator replace-typography-declarations <path>
 ```
 
+### `replace-border-declarations`
+
+Replace length values (`px`, `rem` and `rem()`) in border declarations (`border`, `border-width`, and `border-radius`) with the corresponding Polaris [shape](https://polaris.shopify.com/tokens/shape) token.
```
Comment (lgriffee) [inline polaris-migrator/README.md]:
```suggestion
Replace lengths and functions (`px`, `rem`, `rem()`, `border()`, `border-width()`, `border-radius()`) in border declarations (`border`, `border-width`, and `border-radius`) with the corresponding Polaris [shape](https://polaris.shopify.com/tokens/shape) token.
```

File: .changeset/warm-cows-own.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-migrator': minor
+---
+
+Add `replace-border-declarations` migration
```
Comment (chloerice) [inline .changeset/warm-cows-own.md]:
```suggestion
Added the `replace-border-declarations` migration
```

**Review by sam-b-rose — APPROVED** [review]
🤩 💯  border line failures are no match for this migration, nice work!

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**